### PR TITLE
Pull out some common code snippets for custom struct marshalling tests

### PR DESCRIPTION
**Pull request by elinor-fung**
Just tried to reorganize some of the snippets we have for custom struct marshalling unit tests. No functional or test coverage change.

### CODE REVIEW

**Review by jkoritzinsky — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs [hunk 1]
```diff
@@ -635,37 +635,121 @@ partial class Test
...
+{preDeclaration}
+partial class Test
+{{
+    [LibraryImport(""DoesNotExist"")]
+    public static partial {returnType} Method({parameterType} p);
+}}";
+
+        public static class CustomStructMarshalling
+        {
+            public static string NonBlittableUserDefinedType(bool defineNativeMarshalling = true) => @$"
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs:682]:
```suggestion
            public static string NonBlittableUserDefinedType(bool defineNativeMarshalling = true) => $@"
```

File: src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs [hunk 2]
```diff
@@ -635,37 +635,121 @@ partial class Test
...
 {(defineNativeMarshalling ? "[NativeMarshalling(typeof(Native))]" : string.Empty)}
+[StructLayout(LayoutKind.Sequential)]
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs:684]: This is the default for `struct`s. Do we need to add this for some reason?

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs (hunk 2 above)
Comment (elinor-fung) [reply src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs:684]: This was in the existing snippets - let me try removing it.

Comment (elinor-fung) [reply src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CodeSnippets.cs:684]: Looks like this we get `CS0649` without it... We can probably add that to our list of allowed diagnostics in `AssertPreSourceGeneratorCompilation`.

**Pull request merged**